Let user specify bounding box

meggart opened the pull request:
I factored this into a separate branch because this would change the structure of the cube_config. Currently, when one wants to create a regional cube, one has to specify the parameters `grid_x0` and `grid_y0` which are the offsets of a hypothetical global grid with the specified resolution. This involves some calulculation on the user side when usually the bounding box is known. 

In addition, for very high resolutions (the Sentinel use case by @felixcremer) roundoff-errors get significant and the resulting coordinates were not correct. 

This branch would let the user specify a bounding box as well as the grid_height and grid_width to specifiy the coordinates. Just wanted to get a general opinion by @forman if this is supported or if there was a specific reason for the old behavior.

meggart committed:
let user specify bounding box

felixcremer commented:
Thanks, @meggart. 
I have tried this with my Sentinel-1 data and it works perfectly.
This is the config that I have used. 
model_version = '2.0.2'
lon0=-98.6871269689448098 # eastern boundary
lon1=-98.4947078350864160 # western boundary
lat0=20.6917955161275380 # northern boundary
lat1=20.5106951548490493 # southern boundary
spatial_res=0.000179663
temporal_res = 2
grid_width = 1071
grid_height = 1008
start_time = datetime.datetime(2014, 1, 1, 0, 0)
end_time = datetime.datetime(2020, 1, 1, 0, 0)
ref_time = datetime.datetime(2014, 1, 1, 0, 0)
calendar = 'gregorian'
file_format = 'NETCDF4_CLASSIC'
compression = True
chunk_sizes = (365,30,30) 
You just have to make sure that you are setting all four of the lon and lat variables correctly, because otherwise it will build silently after the old scheme. This should be documented.

Commit:
Remove check for lon and lat values

I want to build regional ESDL datacubes with data which is in a regional
projection like for example UTM.
It seems to work, when I remove the lat and lon value checks, because
these are only applicable in EPSG:4326.
This change needs the boundingbox branch.

meggart committed:
Merge pull request #79 from felixcremer/otherprojections

Allow to build cubes with different coordinate reference systems

meggart commented:
Merging this into the `zarrcreate` branch to have a single new reference
